From 173f465a4506f969aec5815f47c8fb2453f5a63e Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Tue, 17 Aug 2021 13:43:37 +0300
Subject: [PATCH 15/19] s32gen1: siul2: Fix static code issues

commit 173f465a4506f969aec5815f47c8fb2453f5a63e from
https://source.codeaurora.org/external/autobsps32/linux

1. Add 'static' keyword to members used only in this file.
2. Fix type mismatch in return data of function 'common_regmap_init'.

Upstream-Status: Pending

Issue: ALB-7517
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/gpio/gpio-siul2-s32gen1.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/gpio/gpio-siul2-s32gen1.c b/drivers/gpio/gpio-siul2-s32gen1.c
index b484acebab19..2606719e2926 100644
--- a/drivers/gpio/gpio-siul2-s32gen1.c
+++ b/drivers/gpio/gpio-siul2-s32gen1.c
@@ -478,8 +478,8 @@ static const struct regmap_config siul2_regmap_conf = {
 	.cache_type = REGCACHE_FLAT,
 };
 
-struct regmap *common_regmap_init(struct platform_device *pdev,
-				  struct regmap_config *conf, const char *name)
+static struct regmap *common_regmap_init(struct platform_device *pdev,
+					 struct regmap_config *conf, const char *name)
 {
 	struct resource *res;
 	void __iomem *base;
@@ -490,8 +490,7 @@ struct regmap *common_regmap_init(struct platform_device *pdev,
 	size = resource_size(res);
 	base = devm_ioremap(dev, res->start, size);
 	if (IS_ERR(base))
-		return base;
-
+		return ERR_PTR(-ENOMEM);
 
 	conf->val_bits = conf->reg_stride * 8;
 	conf->max_register = size - conf->reg_stride;
@@ -962,13 +961,13 @@ static struct platform_driver siul2_gpio_driver = {
 	.probe		= siul2_gpio_probe,
 };
 
-int siul2_gpio_init(void)
+static int siul2_gpio_init(void)
 {
 	return platform_driver_register(&siul2_gpio_driver);
 }
 module_init(siul2_gpio_init);
 
-void siul2_gpio_exit(void)
+static void siul2_gpio_exit(void)
 {
 	platform_driver_unregister(&siul2_gpio_driver);
 }
-- 
2.17.1

